mlrun/db/httpdb.py

```diff
@@ -915,16 +926,20 @@ def del_runs(self, name=None, project=None, labels=None, state=None, days_ago=0)
 
         :param name: Name of the task which the runs belong to.
         :param project: Project to which the runs belong.
-        :param labels: Filter runs that are labeled using these specific label values.
+        :param labels: Filter runs that are labeled using these specific label values. This can be provided as:
+            - A dictionary in the format {"label": "value"} to match specific label key-value pairs.
+            - A list of strings formatted as "label=value" to match specific label key-value pairs.
+            - A list of strings formatted as "label" to match entities with the specified label key only.
         :param state: Filter only runs which are in this state.
         :param days_ago: Filter runs whose start time is newer than this parameter.
         """
 
         project = project or config.default_project
+        labels = self._validate_labels(labels)
```

Comment:
my main issue with this is that now if I add a new endpoint with labels param I need to remember to use this validator but I don't see a way around it that I like ATM.
Let's rename it to `_parse_labels` and comment that we do this to support providing dict from the SDK while it's not supported in the endpoints.
I think this clarifies that although we do validation here, the actual goal is to support dict.